zkevm-circuits/src/evm_circuit/util/memory_gadget.rs

```diff
@@ -173,23 +290,32 @@ impl<F: FieldExt, const MAX_QUAD_COST_IN_BYTES: usize>
         &self,
         region: &mut Region<'_, F>,
         offset: usize,
-        curr_memory_size: MemorySize,
-        address: Address,
-    ) -> Result<(MemorySize, u128), Error> {
+        curr_memory_size: u32,
+        addresses: [u64; N],
+    ) -> Result<(u64, u64), Error> {
         // Calculate the active memory size
-        let address_memory_size =
-            self.address_memory_size.assign(region, offset, address)?;
+        let memory_sizes = self
+            .memory_sizes
+            .iter()
+            .zip(addresses.iter())
+            .map(|(memory_size, address)| {
+                memory_size.assign(region, offset, *address)
```

Comment:
It's a bit strange that within the same region each size item is placed at same offset. 
Whouldn't be needed to update the offset after each assignment? 

Just a question. Haven't checked this in depth.

Reply:
The `assign` function might answer you question. Currently we don't care about `column` or `rotation` too much, to boost development and explore as much as possible, so the `column` and `rotation` are stored directly inside the struct `Cell`, for correct assignment.

Reply:
Yep, now I see the approach and definitely makes all the work much more easy!